torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp

```diff
@@ -248,7 +248,7 @@ c10::optional<int64_t> Val::getInt() const {
 }
 
 c10::optional<double> Val::getDouble() const {
-  if (isConstScalar() && isAnInt()) {
+  if (isConstScalar() && isADouble()) {
     if (this->getValType() == ValType::Scalar) {
       if (this->isA<Double>()) {
```

Comment:
Should remove these two lines above.

Reply:
Which lines are you referring to? Lines 251 and 252? Wouldn't we still need to have `isConstScalar()`?